documentcloud/documents/models/document.py

```diff
@@ -445,12 +459,108 @@ def set_page_text(self, page_text_infos):
         file_contents.append(json.dumps(json_text).encode("utf-8"))
 
         # upload the text to S3
-        logger.info("[SET PAGE TEXT] upload %d", self.pk)
+        logger.info(
+            "[SET PAGE TEXT] upload %d - %d mb",
+            self.pk,
+            sum(len(i) for i in file_contents) / 1000 / 1000,
+        )
+        for name, contents in zip(file_names, file_contents):
+            logger.info(
+                "[SET PAGE TEXT] upload %d - %s: %d mb",
+                self.pk,
+                name,
+                len(contents) / 1000 / 1000,
+            )
         # reverse the lists to upload the larger files first
         storage.async_upload(file_names[::-1], file_contents[::-1], access=self.access)
 
         return json_text
 
+    def _set_page_positions(self, pages, file_names, file_contents):
```

Comment:
Strong test candidate. Is the page overlay added to the correct page and in the correct position?